kolibri/core/content/api.py

```diff
@@ -1310,6 +1315,79 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(ValuesViewset):
+    pagination_class = OptionalPageNumberPagination
+
+    queryset = ContentDownloadRequest.objects.all()
+
+    field_map = {
+        "source_id": "self.request.user.id",
+    }
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+    )
+
+    def filter_queryset(self, queryset):
+        return queryset.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                )
+            )
+        ).filter(has_removal=False)
+
+    def create(self, request, *args, **kwargs):
+        # if there is an existing deletion request, delete the deletion request
+        deletion_request = ContentRemovalRequest.objects.filter(
+            contentnode_id=self.request["contentnode_id"],
+            source_id=self.request.user.id,
+        )
+
+        if deletion_request:
```

Comment:
To avoid evaluating the entire queryset in the weird circumstance where there are hundreds of Removal requests that meet this filter, you can do:
```
if deletion_request.exists():
```
which will just do a check that there are any with an exists query rather than loading all the requests from the database.